### DISCUSSION TITLE
traefik suddenly not working - how to upgrade from the command line?

### DISCUSSION THREAD

**Discussion by TimHeckel** [kube-hetzner/terraform-hcloud-kube-hetzner#264] [Q&A] [answered]
Hey all - so I think k3 automatically was upgrading traefik when something failed during install. Typically I can do a helm "upgrade" command via lens to get traefik back, but it's unresponsive - is there a helm upgrade command that targets the traefik version that is being used? The app version is listed as 2.6.2. Would love any help here...thanks so much.

**Comment by mysticaltech** [comment, marked as answer]
@TimHeckel Please look in the Traefik template `traefik_config.yaml.tpl`; it should have a spec of `failurePolicy: abort`, which should protect the cluster against failed Traefik upgrades. But maybe the version of Kube-Hetzner you have does not have it. 

So just apply the most recent `traefik_config.yaml.tpl` found in the repo, transfer it to a proper yaml format first manually, and if you want to read your current one via `kubectl get helmchartconfig` in the proper namespace.

**Reply by TimHeckel** [reply]
Thank you! That worked. I transferred the values manually to an install file and ran

```
kubectl apply -f traefik_install.yaml -n kube-system
```

My version of kube-hetzner does not have `failurePolicy: abort` set I think. I haven't had time to upgrade to the latest, but it's on my list.

Thanks again!

**Comment by mysticaltech** [comment]
Awesome to hear @TimHeckel, and to upgrade, the easiest path is just to deploy a new cluster and migrate everything with a sweet tool from VMWare called Velero; more about this here https://velero.io/docs/v1.9/migration-case/.
